spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/gson/GsonProperties.java

```diff
@@ -0,0 +1,195 @@
+package org.springframework.boot.autoconfigure.gson;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.LongSerializationPolicy;
+import com.google.gson.annotations.Expose;
+import com.google.gson.Gson;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import java.util.List;
+
+/**
+ * Configuration properties to configure Jackson.
+ *
+ * @author Ivan Golovko
+ * @since 2.0.0
+ */
+@ConfigurationProperties(prefix = "spring.gson")
+public class GsonProperties {
+
+    /**
+     * Configures {@link Gson} to enable versioning support.
+     */
+    private Double version;
```

Comment:
Do we want to expose all that? You are introducing a customizer so you're giving the ability to a user to pretty much change the builder any way they like. I'd focus on the most used properties.